cl_sii/dte/data_models.py

```diff
@@ -466,6 +468,168 @@ def as_dte_data_l1(self) -> DteDataL1:
+    ))
+)
+class DteXmlReferencia:
+    """
+    Data in XML element ``Referencia`` in an DTE XML doc.
+
+    > Identificacion de otros documentos Referenciados por Documento
+
+    DTE doc XML element: 'Documento//Referencia'
+
+    .. note:: An XML DTE document includes none or up to 40 "Referencia" elements.
+
+    .. seealso::
+        XML schema of ``{http://www.sii.cl/SiiDte}/DTE/Documento/Referencia`` in 'DTE_v10.xsd' at
+        https://github.com/cl-sii-extraoficial/archivos-oficiales/blob/master/src/code/dte/README.md
+
+    """
+
+    ###########################################################################
+    # Fields
+    ###########################################################################
+
+    numero_linea_ref: int
+    """
+    Sequential line  number of the "referencia". Must be an integer between 1 and 40 inclusive
+
+    > Numero Secuencial de Linea de Referencia
+
+    DTE doc XML element: '..//Documento//Referencia//NroLinRef'
+    """
+
+    tipo_documento_ref: str
+    """
+    Kind of the document of "Referencia". Length must be >= 1 and <= 3
+
+    > Tipo de Documento de Referencia
+
+    DTE doc XML element: '..//Documento//Referencia//TpoDocRef'
+
+    .. note::
+        This field accepts any of the elements of the class:`TipoDocumentoEnum` or
+        an alphanumeric to refer to non-tax documents (in this case, validation does not apply)
+    """
+
+    folio_ref: str
+    """
+    The folio of the document referred to.
+
+    > Identificación del documento de referencia.
+
+    DTE doc XML element: '..//Documento//Referencia//FolioRef'
+    """
+
+    fecha_ref: date
+    """
+    The 'fecha_emision' of the document referred to.
+
+    > Fecha del documento de referencia
+
+    DTE doc XML element: '..//Documento//Referencia//FchRef'
+    """
+
+    ind_global: Optional[int] = None
+    """
+    Whether a set of documents of the same kind is referenced.
+
+    > Documento afecta a un número de más de 20 documentos del mismo `tipo_documento_ref`
+    > Se explicita la razón en `Razón Referencia`
+
+    DTE doc XML element: '..//Documento//Referencia//IndGlobal'
+    """
+
+    rut_otro: Optional[Rut] = None
+    """
+    The RUT of the "emisor" of the document referred to.
+
+    > RUT otro contribuyente
+
+    DTE doc XML element: '..//Documento//Referencia//RUTOtr'
+
+    .. note::
+        > Sólo si el documento de referencia es de tipo tributario y fue emitido
+        > por otro contribuyente
+    """
+
+    codigo_ref: Optional[CodigoReferenciaEnum] = None
+    """
+    The type of use for the reference
+
+    > Tipo de Uso de la Referencia
+
+    DTE doc XML element: '..//Documento//Referencia//CodRef'
+    """
+
+    razon_ref: Optional[str] = None
+    """
+    The reason the document is being referenced
+
+    > Razon Explicita por la que se Referencia el Documento
+
+    DTE doc XML element: '..//Documento//Referencia//RazonRef'
+    """
+
+    def __post_init__(self) -> None:
```

Comment:
Since this is a Pydantic dataclass, we might as well leverage Pydantic's validators. @jtrh suggestions?

Reply:
> [...] leverage Pydantic's validators. @jtrh suggestions?

- Refactor all validations done in `__post_init__` as Pydantic validators. Example: https://github.com/fyntex/lib-cl-sii-python/blob/fd39c62d2c7fedfd7e044bfbbab3a0f19531e6db/cl_sii/rtc/data_models.py#L37-L47 and https://github.com/fyntex/lib-cl-sii-python/blob/fd39c62d2c7fedfd7e044bfbbab3a0f19531e6db/cl_sii/rtc/data_models.py#L388-L392.
- Type checks using `isinstance` are not needed because they're automatically performed by Pydantic.